clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureStorageClient.groovy

```diff
@@ -39,43 +39,47 @@ class AzureStorageClient {
   static List<AzureCustomVMImage> getCustomImages(List<AzureCustomImageStorage> imageStorageList) {
     def vmImages = new ArrayList<AzureCustomVMImage>()
 
-    imageStorageList.each {AzureCustomImageStorage storage ->
-      try {
-        ArrayList<String> blobDirectoryList = new ArrayList<String>()
-
-        // Retrieve storage account from connection-string.
-        CloudStorageAccount storageAccount = CloudStorageAccount.parse(storage.scs)
-
-        // retrieve the blob client.
-        CloudBlobClient blobClient = storageAccount.createCloudBlobClient()
-        String dirDelimiter = blobClient.getDirectoryDelimiter()
-        blobDirectoryList.addAll(storage.blobDir.split(dirDelimiter))
-        def container = blobClient.getContainerReference(blobDirectoryList.remove(0))
-
-        if (container) {
-          if (blobDirectoryList.size() > 0) {
-            def dir = blobDirectoryList.remove(0)
-            def blob = container.getDirectoryReference(dir)
-
-            while (blobDirectoryList.size() > 0) {
-              dir = blobDirectoryList.remove(0)
-              blob = blob.getDirectoryReference(dir)
-            }
+    imageStorageList?.each {AzureCustomImageStorage storage ->
+      if (storage && storage.scs && storage.blobDir && storage.osType) {
```

Comment:
Ok as-is, but may read better as: `if (storage?.scs && storage?.blobDir && storage?.osType)`.
Or maybe not.
I can live with it either way.